app-frontend/src/app/components/scenes/sceneItem/sceneItem.module.js

```diff
@@ -58,9 +59,21 @@ class SceneItemController {
         if (changes.repository && changes.repository.currentValue) {
             this.repository = changes.repository.currentValue;
             if (this.scene.sceneType === 'COG') {
+                let redBand = _.findIndex(
+                    this.datasource.bands, (x) => x.name.toLowerCase() === 'red');
+                let greenBand = _.findIndex(
+                    this.datasource.bands, (x) => x.name.toLowerCase() === 'green');
+                let blueBand = _.findIndex(
+                    this.datasource.bands, (x) => x.name.toLowerCase() === 'blue');
+                let atLeastThreeBands = this.datasource.bands.length >= 3;
                 this.sceneService.cogThumbnail(
                     this.scene.id,
-                    this.authService.token()
+                    this.authService.token(),
+                    128,
+                    128,
+                    Math.max(redBand, atLeastThreeBands ? 0 : 0),
+                    Math.max(greenBand, atLeastThreeBands ? 1 : 0),
+                    Math.max(blueBand, atLeastThreeBands ? 2 : 0)
```

Comment:
Ah, ok, I think I figured out what caused the broken thumbnail issue for the above mentioned testing scene -- it is because of these three `Match.max(...)`, which returned 2, 1, 2, respectively in the testing case. I would do something like below instead to avoid it:
```
let atLeastThreeBands = this.datasource.bands.length >= 3;
let foundbands = Math.min(redBand, greenBand, blueBand) >= 0;
let bands = {};
if (atLeastThreeBands) {
    bands.red = foundbands ? redBand : 0;
    bands.green = foundbands ? greenBand : 1;
    bands.blue = foundbands ? blueBand : 2;
 } else {
    bands.red = 0;
    bands.green = 0;
    bands.blue = 0;
}
```
But I am sure you will have more concise ways.

Reply:
I'll take a look but that's pretty concise as is :ok_hand: